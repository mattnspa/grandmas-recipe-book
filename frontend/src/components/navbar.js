import { useState } from "react";
import Container from "react-bootstrap/Container";
import Nav from "react-bootstrap/Nav";
import Navbar from "react-bootstrap/Navbar";
import { Link } from "react-router-dom";

import ImgPath from "./nav-icon-light.svg";
import { SearchBar } from './searchbar';

export const NavbarComponent = props => {
  const { submitButton, location } = props;
  const [activeRoute, setActiveRoute] = useState(location);

  const routes = {
    home: "/",
    recipes: "/recipes",
    info: "/info"
  };

  return (
    <Navbar bg="dark" variant="dark" >
      <Container>
      <Navbar.Brand reloadDocument={activeRoute === routes.home ? true : false}
        as={Link} 
        to={routes.home}
        onClick={() => {setActiveRoute(routes.home)}}>
          <img
            src={ImgPath}
            width="30"
            height="30"
            className="d-inline-block align-top"
            alt="React Bootstrap logo" />
        </Navbar.Brand>
        <Nav className="me-auto">
          <Nav.Link reloadDocument={activeRoute === routes.recipes ? true : false}
            as={Link} 
            to={routes.recipes} 
            active={activeRoute === routes.recipes}
            onClick={() => {setActiveRoute(routes.recipes)}}>
              Recipes
          </Nav.Link>
          <Nav.Link reloadDocument={activeRoute === routes.info ? true : false}
            as={Link} 
            to={routes.info} 
            active={activeRoute === routes.info}
            onClick={() => {setActiveRoute(routes.info)}}>
              Info
          </Nav.Link>
        </Nav>
        {activeRoute === routes.recipes && <SearchBar submitButton={submitButton}/>}
      </Container>
    </Navbar>
  );
}