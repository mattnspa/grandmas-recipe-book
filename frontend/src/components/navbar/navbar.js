import Container from "react-bootstrap/Container";
import Nav from "react-bootstrap/Nav";
import Navbar from "react-bootstrap/Navbar";
import { Link } from "react-router-dom";

import ImgPath from "./nav-icon-light.svg";
import { SearchBar } from '../search/searchbar';

export const NavbarComponent = props => {
  const { activeRoute, setActiveRoute, submitButton } = props;
  
  const routes = {
    home: "/",
    recipes: "/recipes",
    info: "/info"
  };

  const handleClick = (route) => {
    setActiveRoute(route);
  }

  return (
    <Navbar bg="dark" variant="dark" >
      <Container>
      <Navbar.Brand reloadDocument={activeRoute === routes.home ? true : false}
        as={Link} 
        to={routes.home}
        onClick={() => {handleClick(routes.home)}}>
          <img
            src={ImgPath}
            width="30"
            height="30"
            className="d-inline-block align-top"
            alt="React Bootstrap logo" />
        </Navbar.Brand>
        <Nav className="me-auto">
          <Nav.Link reloadDocument={activeRoute === routes.recipes ? true : false}
            as={Link} 
            to={routes.recipes} 
            active={activeRoute === routes.recipes}
            onClick={() => {handleClick(routes.recipes)}}>
              Recipes
          </Nav.Link>
          <Nav.Link reloadDocument={activeRoute === routes.info ? true : false}
            as={Link} 
            to={routes.info} 
            active={activeRoute === routes.info}
            onClick={() => {handleClick(routes.info)}}>
              Info
          </Nav.Link>
        </Nav>
        {activeRoute === routes.recipes && <SearchBar submitButton={submitButton}/>}
      </Container>
    </Navbar>
  );
}