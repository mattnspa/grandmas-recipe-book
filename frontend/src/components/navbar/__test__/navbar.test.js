import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import { NavbarComponent } from "../navbar";



describe("Navbar component is loaded", () => {

  afterEach(cleanup); 

  const homePageText = "home page"
  const infoPageText = "info page"
  const recipesPageText = "recipes page"

  const renderWithRouter = (component) => {
    const history = createMemoryHistory()
    return { 
      ...render (
      <BrowserRouter history={history}>
          {component}
          <Routes>
            <Route path="/" element={<h1>{homePageText}</h1>} />
            <Route path="/info" element={<h1>{infoPageText}</h1>} />
            <Route path="/recipes" element={<h1>{recipesPageText}</h1>}/>
          </Routes>
      </BrowserRouter>
      )
    }
  };

  it("renders correctly", () => {
    // given
    const route = '/';
    const setMockActiveRoute = jest.fn();
    const mockButton = jest.fn();

    // when
    const {getByTestId} = renderWithRouter(
      <NavbarComponent 
        activeRoute={route}
        setActiveRoute={setMockActiveRoute}
        submitButton={mockButton} />);
      
    // then
    const navbar = getByTestId('navbar')
    const homeLink = getByTestId('home-link')
    const recipesLink = getByTestId('recipes-link')
    const infoLink = getByTestId('info-link')

    expect(navbar).toContainElement(homeLink);
    expect(recipesLink).toContainElement(recipesLink);
    expect(infoLink).toContainElement(infoLink);

  });

  it("snapshot is the same", () => {
    // given
    const route = '/';
    const setMockActiveRoute = jest.fn();
    const mockButton = jest.fn();

    // when
    const { asFragment } = renderWithRouter(
      <NavbarComponent
        activeRoute={route}
        setActiveRoute={setMockActiveRoute}
        submitButton={mockButton} />);
        
    // then
    expect(asFragment(<NavbarComponent />)).toMatchSnapshot()
  });

  it("should navigate to the home page, then the info page, then the recipes page", () => {
    // given
    const route = '/';
    const setMockActiveRoute = jest.fn();
    const mockButton = jest.fn();

    // when
    const {container, getByTestId} = renderWithRouter(
      <NavbarComponent 
        activeRoute={route}
        setActiveRoute={setMockActiveRoute}
        submitButton={mockButton} />);

      fireEvent.click(getByTestId('home-link'));

      // then
      expect(container.innerHTML).toMatch(homePageText)
      // when
      fireEvent.click(getByTestId('info-link'));
      // then
      expect(container.innerHTML).toMatch(infoPageText)
      // when
      fireEvent.click(getByTestId('recipes-link'));
      // then
      expect(container.innerHTML).toMatch(recipesPageText)
  });
});