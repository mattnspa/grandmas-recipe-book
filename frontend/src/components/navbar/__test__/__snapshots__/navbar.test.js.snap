// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component is loaded snapshot is the same 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand navbar-dark bg-dark"
    data-testid="navbar"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        data-testid="home-link"
        href="/"
      >
        <img
          alt="React Bootstrap logo"
          class="d-inline-block align-top"
          height="30"
          src="nav-icon-light.svg"
          width="30"
        />
      </a>
      <div
        class="me-auto navbar-nav"
      >
        <a
          class="nav-link"
          data-testid="recipes-link"
          href="/recipes"
        >
          Recipes
        </a>
        <a
          class="nav-link"
          data-testid="info-link"
          href="/info"
        >
          Info
        </a>
      </div>
    </div>
  </nav>
  <h1>
    home page
  </h1>
</DocumentFragment>
`;
